graphx/src/main/scala/org/apache/spark/graphx/Pregel.scala

```diff
@@ -154,7 +169,9 @@ object Pregel extends Logging {
       // count the iteration
       i += 1
     }
-    messages.unpersist(blocking = false)
+    messageCheckpointer.unpersistDataSet()
```

Comment:
Sorry, I don't understand this change. Why do we replace

```scala
messages.unpersist(blocking = false)
```

with

```scala
messageCheckpointer.unpersistDataSet()
```

Especially because this adds a new public method to `PeriodicCheckpointer` that no other code has needed before.